repositories/user_character.go

```diff
@@ -0,0 +1,45 @@
+package repositories
+
+import (
+	"database/sql"
+
+	"github.com/ShinnosukeSuzuki/techtrain-mission-ca-tech-dojo-golang/models"
+)
+
+// リポジトリ構造体を定義
+type UserCharacterRepository struct {
+	db *sql.DB
+}
+
+// リポジトリのコンストラクタ
+func NewUserCharacterRepository(db *sql.DB) UserCharacterRepository {
+	return UserCharacterRepository{db: db}
+}
+
+// userのtokenに一致するキャラクターを取得する
+func (r *UserCharacterRepository) GetUserCharacterList(token string) ([]models.UserCharacter, error) {
+	const sqlSelectCharacterByUserID = `
+		SELECT users_characters.id, users_characters.character_id, characters.name
+		FROM users_characters
+		JOIN users ON users_characters.user_id = users.id
+		JOIN characters ON users_characters.character_id = characters.id
+		WHERE users.token = ?;
```

Comment:
JOINするときは、エイリアスを定義して短くすることが多いです。

```suggestion
		SELECT uc.id, uc.character_id, c.name
		FROM users_characters as uc
		JOIN users as u ON uc.user_id = u.id
		JOIN characters as c ON uc.character_id = c.id
		WHERE u.token = ?;
```

Reply:
https://github.com/ShinnosukeSuzuki/techtrain-mission-ca-tech-dojo-golang/commit/11ebf7a79f4b44ec8ecbccbc04e4fa60d8f9d680
上記コミットで対応しました。